src/weapon/w_038.c

```diff
@@ -70,7 +70,122 @@ static void EntityWeaponAttack(Entity* self) {
     }
 }
 
-INCLUDE_ASM("weapon/nonmatchings/w_038", func_ptr_80170004);
+extern s32 D_10E000_8017B844;
+
+s32 func_ptr_80170004(Entity* self) {
+    Primitive* prim;
+    s32 i;
+    s16 x, y;
+    s16 newX;
+    s16 newY;
+    s16 xOffset, yOffset;
+
+#ifdef VERSION_PSP
+    s16 params;
+    params = (self->params & 0x7fff) >> 8; // unused
```

Comment:
Does this not get automatically optimized out on ps1?

Reply:
It will. Let me take that out.